libs/langchain/langchain/llms/promptguard.py

```diff
@@ -0,0 +1,105 @@
+import logging
+from typing import Any, Dict, List, Optional
+
+from pydantic import Extra, root_validator
+
+from langchain.callbacks.manager import CallbackManagerForLLMRun
+from langchain.llms.base import LLM
+from langchain.prompts.base import StringPromptValue
+from langchain.utils import get_from_dict_or_env
+
+logger = logging.getLogger(__name__)
+
+
+class PromptGuardLLMWrapper(LLM):
+    """
+    A LLM wrapper that uses the PromptGuard to sanitize the prompt before
+    passing it to the LLM, and desanitize the response after
+    getting it from the LLM.
+
+    To use, you should have the `promptguard` python package installed,
+    and the environment variable `PROMPTGUARD_API_KEY` set with
+    your access token, or pass it as a named parameter to the constructor.
+
+    Example:
+        .. code-block:: python
+
+            prompt_guard_llm = PromptGuardLLM(llm=ChatOpenAI())
+    """
+
+    llm: Any
+    """The LLM to use."""
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that the promptguard api key and python package exists
+        in environment."""
+        token = get_from_dict_or_env(
+            values, "promptguard_api_key", "PROMPTGUARD_API_KEY"
+        )
+        if token is None:
+            raise ValueError(
+                "Could not find PROMPTGUARD_API_KEY in environment. "
+                "Please set it to your PromptGuard api key."
+            )
+        try:
+            import promptguard as pg
+
+            assert pg.__package__ is not None
+        except ImportError:
+            raise ImportError(
+                "Could not import the `promptguard` python package. "
+                "Please install it with `pip install promptguard`."
+            )
+        return values
+
+    def _call(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> str:
+        """Call out to PromptGuard to do sanitization and desanitization
+        before and after running LLM.
+
+        This is an override of the base class method.
+
+        Args:
+            prompt: The prompt to pass into the model.
+        Returns:
+            The string generated by the model.
+        Example:
+            .. code-block:: python
+                response = prompt_guard_llm("Tell me a joke.")
```

Comment:
Use the usual function documentation style.

Reply:
I assume this is a match of langchain's style - if not please make sure that we use that style (as that matters more than matching our style)

Reply:
Yes the format is copied from other class. However, I do notice there are different docstring format in Langchain repo, including this and the one we used.

Reply:
I'll use our style then as that looks clearer.